react-app/src/redux/game/reducer.js

```diff
@@ -0,0 +1,57 @@
+import { calculateWinner } from '../../utils';
+
+const initialState = {
+  history: [
+    {
+      squares: Array(9).fill(null)
+    }
+  ],
+  stepNumber: 0,
+  xIsNext: true,
+  winner: null
+};
+
+const handleClick = (state, squarePosition) => {
```

Comment:
Try to have all this logic in the action, the reducer should be to just update the state